pkg/controller/backend/3scale_reconciler.go

```diff
@@ -546,84 +531,89 @@ func (t *ThreescaleReconciler) processNotDesiredMappingRules(notDesiredList []th
 	return nil
 }
 
-func (t *ThreescaleReconciler) reconcileMatchedMappingRules(matchedList []mappingRuleData) error {
-	for _, data := range matchedList {
-		params := threescaleapi.Params{}
+func (t *ThreescaleReconciler) reconcileMappingRuleWithPosition(desired capabilitiesv1beta1.MappingRuleSpec, desiredPosition int, existing threescaleapi.MappingRuleItem) error {
+	params := threescaleapi.Params{}
 
-		//
-		// Reconcile metric or method
-		//
-		metricID, err := t.backendAPIEntity.FindMethodMetricIDBySystemName(data.spec.MetricMethodRef)
-		if err != nil {
-			return fmt.Errorf("Error reconcile backend mapping rule: %w", err)
-		}
+	//
+	// Reconcile metric or method
+	//
+	metricID, err := t.backendAPIEntity.FindMethodMetricIDBySystemName(desired.MetricMethodRef)
+	if err != nil {
+		return fmt.Errorf("Error reconcile backend mapping rule: %w", err)
+	}
 
-		if metricID < 0 {
-			// Should not happen as metric and method references have been validated and should exists
-			return errors.New("backend metric method ref for mapping rule not found")
-		}
+	if metricID < 0 {
+		// Should not happen as metric and method references have been validated and should exists
+		return errors.New("backend metric method ref for mapping rule not found")
+	}
 
-		if metricID != data.item.MetricID {
-			params["metric_id"] = strconv.FormatInt(metricID, 10)
-		}
+	if metricID != existing.MetricID {
+		params["metric_id"] = strconv.FormatInt(metricID, 10)
+	}
 
-		//
-		// Reconcile delta
-		//
-		if data.spec.Increment != data.item.Delta {
-			params["delta"] = strconv.Itoa(data.spec.Increment)
-		}
+	//
+	// Reconcile delta
+	//
+	if desired.Increment != existing.Delta {
+		params["delta"] = strconv.Itoa(desired.Increment)
+	}
 
-		//
-		// Reconcile last
-		//
-		desiredLastAttribute := false
-		if data.spec.Last != nil {
-			desiredLastAttribute = *data.spec.Last
-		}
+	//
+	// Reconcile last
+	//
+	desiredLastAttribute := false
+	if desired.Last != nil {
+		desiredLastAttribute = *desired.Last
+	}
 
-		if desiredLastAttribute != data.item.Last {
-			params["last"] = strconv.FormatBool(desiredLastAttribute)
-		}
+	if desiredLastAttribute != existing.Last {
+		params["last"] = strconv.FormatBool(desiredLastAttribute)
+	}
 
-		if len(params) > 0 {
-			err := t.backendAPIEntity.UpdateMappingRule(data.item.ID, params)
-			if err != nil {
-				return fmt.Errorf("Error reconcile backend mapping rule: %w", err)
-			}
+	//
+	// Reconcile Position
+	//
+	if desiredPosition != existing.Position {
```

Comment:
Or only recreate if there are unwanted MR

Reply:
I'm not sure I'm following your comments.

With the current implementation we are first deleting existing MRs (https://github.com/3scale/3scale-operator/pull/452/files#diff-bbd3f63420102034bbd1e87eb4896baeR482) in the 3scale API before creating/reconciling the desired ones :thinking: 

Then we create/reconcile based in the position of the array in ascending order

Can you further comment on this?

Reply:
ok I think I get your comments now.

I see the position change as a consequence of deleting an existing MR that is not defined in the CR as expected/correct behavior in the scenario you describe.

Recreating would temporarily remove the rule itself changing the behavior temporarily which we should avoid.

Reply:
Recreate the list of existing MR, not the MR themselves. 

`s/recreate/refetch/`

Reply:
:+1: makes sense. I'll take a look at it

Reply:
Contents updated.

The process now is:
* First delete all the MRs not existing in the CR spec
* In case the number of MRs not existing in the CR spec first gathered was > 0 we refetch them again to provide a more consistent view of the existing MRs after the deletion
* Then process each MR specified in CR spec in ascendent array order. If it does not exist it will be created, if it exists it will be reconciled